### DISCUSSION TITLE
关于quic server 只有一个udp socket的疑问

### DISCUSSION THREAD

**Discussion by MeristemForge** [alibaba/xquic#121] [General]
hihi，
 quic server 目前从代码里 我只看到它创建一个udp socket 去处理所有的事情， 这样如果好多客户端都连同一个server，server岂不是很累，因为它只有一个fd 去和多个客户端进行读写。   

同时我也看了下msquic的实现，我看到msquic server 似乎创建了很多udp socket 并且bind 相同的地址，这也让我好奇，如果多个client发消息到服务器，服务器能负载均衡的分配到每隔socket 处理？ 于是我自己写了个udp server 同时创建两个socket 并且bind 同一个地址和端口， 然后用多个client 发消息，似乎server只有一个fd工作。 

我看msquic 的提交记录似乎 创建多个socket 是为了支持 RSS（接收端缩放），这个是只有硬件支持才能看到效果？

**Comment by adcen0107** [comment]
1. xquic设置的是一个cb去操作socket，这个是可以被替换的，这几个test仅仅是收、发的功能演示，并不是真实环境中client与server的 APP与 socket层IO实现；
2. 我怀疑是你的demo写得有问题，应该使用so_reuseport实现端口复用，从而udp收包被简单负载均衡；
3. rss也是实现简单的lb功能，如2所讲，kernel也支持对udp packet实现这样功能。
